WalletWasabi/Services/SystemAwakeChecker.cs

```diff
@@ -63,43 +78,71 @@ private SystemAwakeChecker(WalletManager walletManager, Func<Task<IPowerSavingIn
 			return new SystemAwakeChecker(walletManager, taskFactory);
 		}
 
-		protected override async Task ActionAsync(CancellationToken cancel)
+		public async Task UpdateAsync()
 		{
-			IPowerSavingInhibitorTask? task = _powerSavingTask;
-
 			if (WalletManager.AnyCoinJoinInProgress())
 			{
-				if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) || RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+				if (_powerSavingTask is null)
 				{
-					if (task is not null)
-					{
-						if (!task.Prolong(Timeout.Add(TimeSpan.FromMinutes(1))))
-						{
-							Logger.LogTrace("Failed to prolong the power saving task.");
-							task = null;
-						}
-					}
+					_powerSavingTask = await TaskFactory!().ConfigureAwait(false);
+				}
 
-					if (task is null)
-					{
-						Logger.LogTrace("Create new power saving prevention task.");
-						_powerSavingTask = await TaskFactory!().ConfigureAwait(false);
-					}
+				if (WalletManager.AnyCoinJoinInProgress()) // WalletManager.AnyCoinJoinInCriticalPhase() will be here
```

Comment:
I believe `WalletManager` should not be in `SystemAwakeChecker` at all.

Reply:
We use it in LOC 83 `if (WalletManager.AnyCoinJoinInProgress())` and there will be an `if (WalletManager.AnyCoinJoinInCriticalPhase())` below that.

Reply:
But i need to talk with @molnard about this PR. Lot of things might change.